### PR TITLE
Parser 6 salena

### PR DESCRIPTION
**Pull request by SalenaAshton**
Parses all types of defined sentences, which include And, Or, Connected, Literal, Not, Imply, Exist and Forall. 
Each set of tests are for simple goal definitions to test that feature only, with the exception of Exist and Forall. 
Exist tests the 'exists' feature combined with AndSentence.
Forall tests the 'forall'feature combined with NotSentence.

### CODE REVIEW

File: lib/parsing/ast.hpp [hunk 1]
```diff
@@ -88,22 +88,28 @@ namespace ast {
...
+    };
+
     struct Domain : x3::position_tagged {
         Name name;
         std::vector<std::string> requirements;
         TypedList<Name> types;
         TypedList<Name> constants;
         std::vector<AtomicFormulaSkeleton> predicates;
-        // std::vector<Action> actions;
+        std::vector<Action> action;
```
Comment (adarshp) [inline lib/parsing/ast.hpp:112]: Can you change `action` to `actions` here to reflect that there are potentially multiple actions associated with a domain?

File: lib/parsing/domain_def.hpp [hunk 2]
```diff
@@ -218,6 +218,37 @@ namespace parser {
                                 +atomic_formula_skeleton >> ')';
     BOOST_SPIRIT_DEFINE(predicates);
 
+
+/***** Start ***** ***** ***** end current stuff ***** ***** ***** *****/
```
Comment (adarshp) [inline lib/parsing/domain_def.hpp:222]: Do we need this comment line? If not, could you remove it?

File: lib/parsing/domain_def.hpp [hunk 3]
```diff
@@ -218,6 +218,37 @@ namespace parser {
...
+    rule<class TAction, ast::Action> const action = "action";
+    auto const action_def = '('
+                            >> lit(":action")
+                            >> name
+                            >> parameters
+                            >> precondition
+                            >> ')';
+    BOOST_SPIRIT_DEFINE(action);
+
+/***** ***** ***** ***** end current stuff ***** ***** ***** *****/
```
Comment (adarshp) [inline lib/parsing/domain_def.hpp:245]: Do we need this comment line?

File: test/test_parser.cpp [hunk 4]
```diff
@@ -76,6 +76,13 @@ BOOST_AUTO_TEST_CASE(test_parser) {
     BOOST_TEST(in(PrimitiveType{"type1"},
                   get<EitherType>(tl.explicitly_typed_lists[0].type)));
 
+    // Test explicitly typed list of variables
```
Comment (adarshp) [inline test/test_parser.cpp:79]: Thanks for adding this test!

File: test/test_parser.cpp [hunk 5]
```diff
@@ -134,7 +141,7 @@ BOOST_AUTO_TEST_CASE(test_parser) {
             ;        ;(walls-built ?s)
```
Comment (adarshp) [inline test/test_parser.cpp:141]: Can you remove the commented out portions here?

**Review by adarshp — CHANGES_REQUESTED** [review]
@SalenaAshton almost there! Could you please do the following:

- address the inline comments I made
- extend the existing domain example in test_parser.cpp with actions
- add some tests to test whether the actions in the domain example are parsed correctly.

Once these changes are made, I'll merge this PR.

File: lib/parsing/ast.hpp (hunk 1 above)
Comment (SalenaAshton) [reply lib/parsing/ast.hpp:112]: Done

**Comment by SalenaAshton** [comment]
outdated. See parser 6.1

**Pull request closed without merging**